Annotate RemoteMediaSessionHelperProxy with MediaPlaybackEnabled

darryl-apple opened the pull request:
#### 4620367fbbe498128b0b97c6bc4d252aa6306e49
<pre>
Annotate RemoteMediaSessionHelperProxy with MediaPlaybackEnabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=284395">https://bugs.webkit.org/show_bug.cgi?id=284395</a>
<a href="https://rdar.apple.com/141235299">rdar://141235299</a>

Reviewed by Sihui Liu.

Annotate RemoteMediaSessionHelperProxy with MediaPlaybackEnabled

* Source/WebKit/GPUProcess/media/ios/RemoteMediaSessionHelperProxy.messages.in:

Canonical link: <a href="https://commits.webkit.org/287679@main">https://commits.webkit.org/287679@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eba5cbc89b9e6e7b7aadd93339f1b764715e4d6a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/80487 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/59494 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/34154 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/85008 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/31469 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/68555 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/7798 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/62894 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/20702 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/83556 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/52993 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/73278 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/43197 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/50296 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/29928 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/71440 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/27957 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/86442 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/7713 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/5452 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/71187 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/7888 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/69115 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/70426 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17538 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/14429 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/13370 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/7675 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/13194 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/7514 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/11033 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/9319 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebKit/GPUProcess/media/ios/RemoteMediaSessionHelperProxy.messages.in:
```diff
@@ -29,10 +29,10 @@ messages -> RemoteMediaSessionHelperProxy {
 #if PLATFORM(WATCHOS)
-    StartMonitoringWirelessRoutes()
-    StopMonitoringWirelessRoutes()
+    [EnabledBy=MediaPlaybackEnabled] StartMonitoringWirelessRoutes()
```
Should the annotation be added to the RemoteMediaSessionHelperProxy instead?

darryl-apple committed:
Annotate RemoteMediaSessionHelperProxy with MediaPlaybackEnabled
https://bugs.webkit.org/show_bug.cgi?id=284395
rdar://141235299

Reviewed by Sihui Liu.

Annotate RemoteMediaSessionHelperProxy with MediaPlaybackEnabled

* Source/WebKit/GPUProcess/media/ios/RemoteMediaSessionHelperProxy.messages.in:

Canonical link: https://commits.webkit.org/287679@main
